### DISCUSSION TITLE
Automation of SQL Patching with DBATools

### DISCUSSION THREAD

**Discussion by sharmakneeraj4** [dataplat/dbatools#7395] [Q&A] [answered]
Update-DbaInstance -ComputerName $Servers -Path '\xyz\SQL2012\CU' -Credential $cred -Version $version -Restart -Confirm:$false

I want to generate a log that can tell if there is any failure in the script. Can someone please suggest how to do that?

**Comment by niphlod** [comment]
isn't the "Notes" property reporting what you're seeking already ?

**Reply by sharmakneeraj4** [reply]
I guess you are talking about messages that come while running this script or I am missing something here?

All those notes are displayed on my profile. So consider a case when my RDP session is disconnected and few servers were not patched with XYZ reason. I need to go through logs and find out reason why servers were not patched.

**Reply by niphlod** [reply]
Update-DbaInstance returns an object for each updated instance that has a property "Notes" with (if they pop up) exceptions that occurred during the update phase. Just save it to a variable/file/whatever and you should be fine.
i.e. 
```
$results = Update-DbaInstance - .......
$results | Export-CliXml path\to\file
```

you can later inspect them by doing

```
$results = Import-CliXml path\to\file
$results
## specifically, you'd need to look at $results.Notes
```

if you want instead a full trace of your script, that contains other functions other than Update-DbaInstance, look at [Start-Transcript](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.host/start-transcript?view=powershell-5.1), or use redirection.

**Reply by sharmakneeraj4** [reply]
Thanks. I will try it out.

**Reply by potatoqualitee** [reply]
In addition, `Get-DbatoolsLog` should help

**Comment by pareshmotiwala** [comment]
Hi Guys(and gals) would this apply Windows patches as well? Somehow we have to apply Windows patches every month on a 100+ servers, some with AG and some not.

**Reply by niphlod** [reply, marked as answer]
nope. look at https://github.com/mgajda83/PSWindowsUpdate
